src/m365/spo/commands/listitem/listitem-retentionlabel-remove.spec.ts

```diff
@@ -0,0 +1,269 @@
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('prompts before removing retentionlabel when confirmation argument not passed (id)', async () => {
+    await command.action(logger, { options: { debug: false, listItemId: 1, webUrl: webUrl, listTitle: listTitle } });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing list item when prompt not confirmed', async () => {
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+    await command.action(logger, {
+      options: {
+        debug: false,
+        listTitle: listTitle,
+        webUrl: webUrl,
+        listItemId: 1
+      }
+    });
+    assert(loggerLogToStderrSpy.notCalled);
+  });
+
+  it('removes the retentionlabel based on listId when prompt confirmed', async () => {
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/lists(guid'${formatting.encodeQueryParameter(listId)}')/items(1)/SetComplianceTag()`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        listId: listId,
+        webUrl: webUrl,
+        listItemId: 1
+      }
+    });
+    assert(loggerLogToStderrSpy.calledWith());
+  });
+
+  it('removes the retentionlabel based on listTitle when prompt confirmed', async () => {
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/lists/getByTitle('${formatting.encodeQueryParameter(listTitle)}')/items(1)/SetComplianceTag()`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        listTitle: listTitle,
+        webUrl: webUrl,
+        listItemId: 1
+      }
+    });
+    assert(loggerLogToStderrSpy.calledWith());
+  });
+
+  it('removes the retentionlabel based on listUrl', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/GetList(@a1)/items(@a2)/SetComplianceTag()?@a1='%2F${formatting.encodeQueryParameter(listUrl)}'&@a2='1'`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        confirm: true,
+        listUrl: listUrl,
+        webUrl: 'https://contoso.sharepoint.com',
+        listItemId: 1
+      }
+    });
+    assert(loggerLogToStderrSpy.calledWith());
```

Comment:
So you're saying here the test case succeeds if the logger spy is called with nothing.

I'd rather we use `await assert.doesNotReject` wrapping the command.action I think.

Or use isNotCalled (or whatever its called) on the loggerLogSpy.